src/test/kotlin/com/vanniktech/rxriddles/tools/RxRule.kt

```diff
@@ -16,14 +16,20 @@ import java.util.concurrent.TimeUnit
  * I didn't do this here though, to ease solving the riddles and keep everything as simple as possible.
  */
 class RxRule(private val testScheduler: TestScheduler = TestScheduler()) : Scheduler(), TestRule {
-  override fun apply(base: Statement, description: Description): Statement {
-    // Override all of the default schedulers so we can use our testScheduler when testing.
-    RxJavaPlugins.setComputationSchedulerHandler { testScheduler }
-    RxJavaPlugins.setIoSchedulerHandler { testScheduler }
-    RxJavaPlugins.setSingleSchedulerHandler { testScheduler }
-    RxJavaPlugins.setNewThreadSchedulerHandler { testScheduler }
+  override fun apply(base: Statement, description: Description): Statement = object : Statement() {
```

Comment:
Can we you revert the `object : Statement`? Or is this necessary?

Reply:
We need it, since `apply` returns `Statement`. We can't return `base` anymore, because now we call `reset()` after the test.